sys/sched_round_robin/sched_round_robin.c

```diff
@@ -82,7 +82,7 @@ void sched_runq_callback(uint8_t prio)
     }
 
     if (_current_rr_priority == prio) {
-        if (sched_runqueue_len2(prio) == 0) {
+        if (sched_runqueue_is_empty(prio)) {
```

Comment:
not sure if that improved the readability

Reply:
I'm pretty sure it did. I wouldn't need to consult the API doc for `sched_runqueue_is_empty()`, but I most certainly need to look up what `sched_runqueue_len2()` is.

If it would be `sched_runqueue_len(prio) == 0`, it would be equally readable. But that would come with a performance penalty.

Btw.: Have you checked the impact on ROM size? I have the feeling that not every compiler is able to optimize `sched_runqueue_len2(prio) == 0` down to the same code `sched_runqueue_is_empty(prio)` generates. If that is true, this might be a point that can convince you of using `sched_runqueue_is_empty()` here.

Reply:
the new version needs 4 extra bytes (native) the (starting with -Og sched_runqueue_len2(prio) == 0 got optimized quiet well) (i only tested gcc)

Reply:
`native` has plenty of ROM and except for `native`, we use `-Os`.

On the Nucleo-F767ZI with `-Os` ROM is reduced from 14444 to 1428 with the new separate functions.

Reply:
the 4 extra bytes went away with an recompilation ; I am investigating

Reply:
Have you compiled with `RIOT_CI_BUILD=1`? Otherwise the RIOT version string depends on the branch name.

But honestly: Do you really thing time to investigate ROM consumption on `native`, or when compiled with `-Og` is well spent? And here, it is for `native` *and* `-Og` ...

Reply:
[core/shed: doc roundrobin -> advance](https://github.com/RIOT-OS/RIOT/pull/16126/commits/f7558630f7fd4fc5e4da3248b15d317d742873b3)
has some extra bytes (native) that are not there before or after the diff to before is just documentation, after has the cardinality functions introduced. with a clean recompilation it is more than 4 bytes in either direction.

the 4 of these extra bytes are an extra push in xtimer and a eax pointer dereference in the runqueue_callback

Reply:
i stop the investiogation now, this commit will be gone anyway, somehow this specific commit is not good while the ones befor and after are different they are the same size and i think they will yield the same performance

Reply:
thanks for the `RIOT_CI_BUILD=1`